src/Transports.Subscriptions.Abstractions/Subscription.cs

```diff
@@ -56,7 +57,31 @@ public void OnCompleted()
             _unsubscribe = null;
         }
 
-        public void OnError(Exception error) => throw new NotImplementedException();
```

Comment:
How did it work before?

Reply:
The subscription execution strategy converted OnError events to OnData events with an ExecutionResult wrapping the error. The PR over there still wraps the error in an ExecutionError but passes it along OnError for it to get wrapped here by this PR.